src/components/TodoItem.tsx

```
@@ -0,0 +1,153 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  isTempTodo?: boolean;
+  onDelete?: (id: number) => void;
+  onUpdate?: (id: number, data: Partial<Todo>) => Promise<void>;
+  isDeleting?: boolean;
+  isUpdating?: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete = () => {},
+  onUpdate,
+  isTempTodo = false,
+  isDeleting = false,
+  isUpdating = false,
+}) => {
+  const { id, title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [editTitle, setEditTitle] = useState(title);
+  const editInputRef = useRef<HTMLInputElement>(null);
+
+  const isLoading = isTempTodo || isDeleting || isUpdating;
+
+  useEffect(() => {
+    if (isEditing && editInputRef.current) {
+      editInputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleToggle = async () => {
+    if (onUpdate && !isTempTodo) {
+      await onUpdate(id, { completed: !completed });
+    }
+  };
+
+  const handleDoubleClick = () => {
+    if (!isTempTodo) {
+      setIsEditing(true);
+      setEditTitle(title);
+    }
+  };
+
+  const handleSave = async () => {
+    const newTitle = editTitle.trim();
+
+    if (newTitle === title) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    if (!newTitle) {
+      if (onDelete) {
+        onDelete(id);
+      }
+
+      return;
+    }
+
+    if (onUpdate) {
+      try {
+        await onUpdate(id, { title: newTitle });
+        setIsEditing(false);
+      } catch {
+        // Error is handled in parent component
+        // Stay in editing mode on error
+      }
```

remove empty catch block